scripts/gdal/gdal_preset.py

```diff
@@ -1,122 +1,141 @@
-from typing import List
+from typing import List, Optional
 
 from linz_logger import get_log
+from scripts.gdal.gdalinfo import gdal_info, GdalInfoBands
 
-GDAL_PRESET_LZW = [
-    "gdal_translate",
-    "-q",
-    "-scale",
+
+# Force the source projection as NZTM EPSG:2193
+NZTM_SOURCE = [   
+    "-a_srs",
+    "EPSG:2193"
+]
+
+# Scale imagery from 0-255 to 0-254 then set 255 as NO_DATA
+# Useful for imagery that does not have a alpha band
+SCALE_254_ADD_NO_DATA = [  
+     "-scale",
     "0",
     "255",
     "0",
     "254",
-    "-a_srs",
-    "EPSG:2193",
     "-a_nodata",
-    "255",
-    "-b",
-    "1",
-    "-b",
-    "2",
-    "-b",
-    "3",
-    "-of",
-    "COG",
-    "-co",
-    "compress=lzw",
-    "-co",
-    "num_threads=all_cpus",
-    "-co",
-    "predictor=2",
-    "-co",
-    "overview_compress=webp",
-    "-co",
-    "bigtiff=yes",
-    "-co",
-    "overview_resampling=lanczos",
-    "-co",
-    "blocksize=512",
-    "-co",
-    "overview_quality=90",
-    "-co",
-    "sparse_ok=true",
+    "255"
 ]
 
-GDAL_PRESET_WEBP = [
-    "gdal_translate",
+BASE_COG = [
+    # ??
     "-q",
-    "-a_srs",
-    "EPSG:2193",
-    "-b",
-    "1",
-    "-b",
-    "2",
-    "-b",
-    "3",
+    # Output to a COG
     "-of",
     "COG",
+    # Tile the image int 512x512px images
     "-co",
-    "compress=webp",
+    "blocksize=512",
+    # Ensure all CPUs are used for gdal translate
     "-co",
     "num_threads=all_cpus",
+    # If not all tiles are needed in the tiff, instead of writing empty images write a null byte
+    # this significantly reduces the size of tiffs which are very sparse
     "-co",
-    "quality=100",
-    "-co",
-    "overview_compress=webp",
+    "sparse_ok=true",
+    # Force everything into big tiff
+    # this converts all offsets from 32bit to 64bit to support TIFFs > 4GB in size
     "-co",
     "bigtiff=yes",
+]
+
+COMPRESS_LZW = [
+    # Compress as LZW
     "-co",
-    "overview_resampling=lanczos",
-    "-co",
-    "blocksize=512",
-    "-co",
-    "overview_quality=90",
+    "compress=lzw",
+    # Predictor two reduces file size
```

Comment:
TODO find a better description as to why predictor two is better.

Reply:
`PREDICTOR=2` stores the difference to the previous value rather than the actual value. `PREDICTOR=1` stores the actual value and `PREDICTOR=3` rearranges bits in float representations to optimise compression for floating point numbers and wouldn't have any effect for integers.